sklearn/ensemble/voting_classifier.py

```diff
@@ -163,6 +175,15 @@ def fit(self, X, y, sample_weight=None):
         if n_isnone == len(self.estimators):
             raise ValueError('All estimators are None. At least one is '
                              'required to be a classifier!')
+
+        if self.flatten_transform == 'default':
+            warnings.warn("'flatten_transform' is changed to False."
+                          " Setting it explicitly will silence this warning",
+                          DeprecationWarning)
+            warnings.warn("'flatten_transform' default value will be "
+                          "changed to True in 0.21.", DeprecationWarning)
+            self.flatten_transform = False
```

Comment:
We shouldn't be modifying parameters. Just handle the default case in transform. In fact it makes sense to only raise this warning in transform as it won't be relevant to the majority of users who only care about prediction